### PR TITLE
Add support for Erlang 17.4.

### PR DESCRIPTION
**Pull request by cmeiklejohn**
Fix namespaced_types; use platform definitions instead of rebar.script.

### CODE REVIEW

File: rebar.config [hunk 1]
```diff
@@ -13,7 +15,7 @@
   {basho_stats, ".*", {git, "git://github.com/basho/basho_stats.git", {tag, "1.0.3"}}},
   {riak_sysmon, ".*", {git, "git://github.com/basho/riak_sysmon.git", {branch, "develop"}}},
   {riak_ensemble, ".*", {git, "git://github.com/basho/riak_ensemble", {branch, "develop"}}},
-  {pbkdf2, ".*", {git, "git://github.com/basho/erlang-pbkdf2.git", {tag, "2.0.0"}}},
+  {pbkdf2, ".*", {git, "git://github.com/basho/erlang-pbkdf2.git", {tag, "develop"}}},
```
Comment (slfritchie) [inline rebar.config]: Chris, there is no such "develop" branch for basho/erlang-pbkdf2 right now.  I can get it to compile with "master", but I dunno if that's the right branch to use there, sorry.

**Comment by slfritchie** [comment]
There's also an R17 types problem for "meck", though it's an indirect one (riak_core's rebar.config doesn't explicitly mention "meck", wheee).  Though that's when compiling riak_core by itself; I haven't tried compiling this branch + riak_kv or riak.

File: rebar.config (hunk 1 above)
Comment (cmeiklejohn) [inline rebar.config]: This should have stayed on 2.0; committed by accident.

**Comment by cmeiklejohn** [comment]
First part in resolving the `meck` dependency issues: https://github.com/basho/exometer_core/pull/3

**Comment by slfritchie** [comment]
Chris also created basho/exometer_core#4

**Comment by cmeiklejohn** [comment]
Tagged 0.8.1p3 of poolboy, 1.0.0-basho2 of exometer_core.

**Comment by slfritchie** [comment]
+1 11e75cdbbc389110caa11d73c6b8c268507c6858

**Comment by slfritchie** [comment]
@borshop merge

**Comment by slfritchie** [comment]
@borshop merge

**Comment by zeeshanlakhani** [comment]
@cmeiklejohn awesome stuff!

**Comment by jadeallenx** [comment]
@cmeiklejohn @slfritchie Thanks for getting this in.

**Pull request merged**